### PR TITLE
Bmulligan/customer viewer data

### PR DESCRIPTION
**Pull request by bmulligan-kg**
https://zube.io/muxinc/mux-ios-sdks/c/4

### CODE REVIEW

File: MUXSDKStats/MUXSDKStatsTests/MUXSDKStatsTests.m [hunk 1]
```diff
@@ -48,11 +48,11 @@ @interface MUXSDKStatsTests : XCTestCase
 @end
 
 @implementation MUXSDKStatsTests
-static NSString *BANDWIDTH = @"BANDWIDTH";
-static NSString *FRAMERATE = @"FRAME-RATE";
-static NSString *X = @"X";
-static NSString *Y = @"Y";
-static NSString *Z = @"Z";
+static const NSString *BANDWIDTH = @"BANDWIDTH";
```
Comment (nidhik) [inline MUXSDKStats/MUXSDKStatsTests/MUXSDKStatsTests.m:51]: +1

File: MUXSDKStats/MUXSDKStatsTests/MUXSDKStatsTests.m [hunk 2]
```diff
@@ -493,13 +532,53 @@ - (void)testupdateCustomerDataWithNullPlayerDataAndVideoData{
     [MUXSDKStats destroyPlayer:playName];
 }
 
+- (void)testUpdateCustomerDataViewerData {
```
Comment (nidhik) [inline MUXSDKStats/MUXSDKStatsTests/MUXSDKStatsTests.m:535]: Great test

File: MUXSDKStats/MUXSDKStats.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -783,6 +783,7 @@
 			baseConfigurationReference = 724162C20BF5162D6FEBFFE5 /* Pods-MUXSDKStatsTests.debug.xcconfig */;
 			buildSettings = {
 				DEVELOPMENT_TEAM = CX6AHWLHM6;
+				ENABLE_BITCODE = YES;
```
[the remark this answers is not in the document]
Comment (bmulligan-kg) [reply MUXSDKStats/MUXSDKStats.xcodeproj/project.pbxproj:786]: this was me debugging a weird build issue... i forgot to remove it afterwards.

File: MUXSDKStats/MUXSDKStats/MUXSDKStats.m [hunk 4]
```diff
@@ -68,11 +78,20 @@ + (void)initSDK {
...
     MUXSDKViewerData *viewerData = [[MUXSDKViewerData alloc] init];
-    NSString *bundleId = [[NSBundle mainBundle] bundleIdentifier];
-    if (bundleId) {
-        [viewerData setViewerApplicationName:bundleId];
+
+    NSString *applicationName = [_customerViewerData viewerApplicationName];
+    if (applicationName == nil) {
+        applicationName = [[NSBundle mainBundle] bundleIdentifier];
     }
+    if (applicationName != nil) {
```
[the remark this answers is not in the document]
Comment (bmulligan-kg) [reply MUXSDKStats/MUXSDKStats/MUXSDKStats.m:91]: this is to preserve the existing behaviour for if [[NSBundle mainBundle] bundleIdentifier] returns nil

[the remark this answers is not in the document]
Comment (nidhik) [reply MUXSDKStats/MUXSDKStats/MUXSDKStats.m:91]: Ah I see

**Review by nidhik — APPROVED** [review]

**Pull request merged**